feat: 예약 기능 구현 (#35)

Lee-sh98 opened the pull request:
## 개요
- close #35 

## 작업사항
- 예약 생성
- 예약 조회
- 예약 수정
- 예약 삭제

Lee-sh98 committed:
feat: 기구별 사용 및 반납 메소드 구현(#35)

EquipmentGym.use
EquipmentGym.restore

Lee-sh98 committed:
feat: 사용 가능한 기구 조회 메소드 구현(#35)

Lee-sh98 committed:
feat: Reservation 빌더 생성자 구현(#35)

Lee-sh98 committed:
feat: Reservation 컨트롤러 구현(#35)

Lee-sh98 committed:
feat: Reservation Request Dto 구현(#35)

Lee-sh98 committed:
feat: Reservation Response Dto 구현(#35)

Lee-sh98 committed:
feat: 예약 생성 기능 구현(#35)

Lee-sh98 committed:
feat: 예약 조회 기능 controller (#35)

Lee-sh98 committed:
feat: 상세한 예약 정보 조회 dto 구현 (#35)

Lee-sh98 committed:
feat: ReservationListResponse 구현 (#35)

Lee-sh98 committed:
feat: 사용자 id를 통한 예약 조회 (#35)

ReservationRepository.findAllByReserverId

Lee-sh98 committed:
feat: 사용자의 예약 정보 조회 기능 구현 (#35)

Lee-sh98 committed:
Merge branch 'develop' of https://github.com/NewFit/NewFit-Backend into feature/35-develop-reservation

Lee-sh98 committed:
feat: 예약 수정 메소드 구현 (#35)

Lee-sh98 committed:
feat: 예약 수정 컨트롤러 구현 (#35)

Lee-sh98 committed:
feat: 예약 수정 requestDto (#35)

Lee-sh98 committed:
feat: 예약 수정 로직 구현 (#35)

Sangwook02 committed:
chore: AuthorityController의 이름을 naming convention에 맞게 수정 (#37) (#41)

Lee-sh98 committed:
feat: 예약 삭제 컨트롤러 구현 (#35)

Lee-sh98 committed:
feat: 예약 삭제 로직 구현 (#35)

Lee-sh98 committed:
refactor: change domain for reservation (#35)

Lee-sh98 committed:
feat: Sl4j 적용 (#35)

Lee-sh98 committed:
merge: pull develop and resolve conflict (#35)

resolve conflict in two files
- ReservationRepository
- ReservationService

Sangwook02 reviewed (commented):
확인 부탁드려요

hoyun06 reviewed (commented):
상욱님이 대부분 다 언급하셔서 그 이외에 궁금한 점 comment 달았습니다!

Lee-sh98 committed:
feat: ReservationRequest validation 강화 (#35)

repetitionNumber에 대해서도 not null validation 적용

Lee-sh98 committed:
feat: Reservation 컨트롤러 유효성 검사 추가 (#35)

예약 수정 기능의 RequestBody 유효성 검사 추가

Lee-sh98 committed:
feat: Reservation 수정 리퀘스트 유효성 검사 (#35)

Future validation 적용

Lee-sh98 committed:
refactor: reservationDetailResponseList 변수화 (#35)

Lee-sh98 committed:
fest: 시간대 중복 확인 메소드 구현(#35)

Lee-sh98 committed:
feat: 예약 정보 조회 컨트롤러 쿼리파라미터 적용(#35)

Lee-sh98 committed:
fix: 예약 생성 및 수정 시 중복 여부 확인, 예약 조회 로직 변경 (#35)

validateReservationOverlap()
- 요청 시간대가 특정 기구(EquipmentGym)의 예약 현황과 겹치면 false 를 반환함

reserve(): 예약 가능한 기구 중 하나에 예약을 진행

update(): 해당 기구의 예약 현황 상 수정 가능하면 예약 수정

listReservation(): 특정 기구의 예약 현황을 조회하도록 수정

Lee-sh98 committed:
fix: 예약 수정 로직 변경 (#35)

해당 기구에서만 수정이 가능하도록 변경

Lee-sh98 committed:
refactor: 예약 수정 로직 간소화 (#35)

Sangwook02 reviewed (commented):
하나 더 확인부탁드립니다.

Lee-sh98 committed:
feat: ReservationRequest validation 수정 (#35)

startAt, endAt: @Future 적용
repetitionNumber: @Min 적용

hoyun06 reviewed (approved):
LGTM
